src/main/java/microsoft/exchange/webservices/data/property/complex/AttachmentCollection.java

```diff
@@ -107,7 +108,8 @@ public FileAttachment addFileAttachment(String fileName) {
    * @return A FileAttachment instance.
    */
   public FileAttachment addFileAttachment(String name, String fileName) {
-    FileAttachment fileAttachment = new FileAttachment(this.owner);
+    FileAttachment fileAttachment = new FileAttachment(
+        this.owner, this.owner.getService().isTraceEnabledFor(TraceFlags.EwsResponse));
```

Comment:
FileAttachment doesn't have a reference to the parent ServiceObject which has all the trace settings info. A boolean arg was added to the FileAttachment constructor so that we can pass in whether or not EwsResponse trace is enabled.